map/framework.cpp

```diff
@@ -1132,12 +1132,13 @@ bool Framework::QueryMayBeSkipped(search::SearchParams const & params,
 
 void Framework::LoadSearchResultMetadata(search::Result & res) const
 {
-  if (res.m_metadata.m_isInitialized)
+  if (res.m_metadata.m_isInitialized || res.GetResultType() != search::Result::RESULT_FEATURE)
```

Comment:
Because it can potentially make code faster. And it’s already set initialized in some cases (see one of search::Result constructor).

But I’m not insisting on it. This code should be refactored and thrown out anyway.

> On Mar 9, 2016, at 22:30, ygorshenin <notifications@github.com> wrote:
> 
> In map/framework.cpp <https://github.com/mapsme/omim/pull/2155#discussion_r55574454>:
> 
> > @@ -1132,12 +1132,13 @@ bool Framework::QueryMayBeSkipped(search::SearchParams const & params,
> >  
> >  void Framework::LoadSearchResultMetadata(search::Result & res) const
> >  {
> > -  if (res.m_metadata.m_isInitialized)
> > +  if (res.m_metadata.m_isInitialized || res.GetResultType() != search::Result::RESULT_FEATURE)
> Disagree. Why metadata should be marked as initialized for non feature results?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2155/files#r55574454>.
>